### PR TITLE
Admin can get / edit / delete Ingredient  

### PR DESCRIPTION
**Pull request by dchima**
## PR Overview
This PR adds features to allow an admin to get, delete, and edit ingredients from the database

Roles
- Admin
- user

Also added
-  admin validations
- routes for get/edit/delete
- created ingredients controllers
- edited swagger docs
- user tests

### Algorithm on PATCH
- request sent to route
- request body contains the edit data: name and units 
- verification and authentications are carried out
- soft validations carried on on body object (no required)
- check if ingredient with name exists in database
- edit ingredient via controller

### Algorithm on GET
- check verifications and authentications
- get all ingredients via controller

### Algorithm on DELETE
- request sent to route
- request body contains the edit data: name and units 
- verification and authentications are carried out
- check if ingredient with name exists in database
- delete ingredient via controller

## Testing
- Clone the repo and cd into the project folder
- Update or copy `.env` file from `.env.sample`
- Checkout to `ft-user-search` branch and run `npm install`
- Run `npm run migrate:reset` && `npm run migrate` to create tables in your 
 database and seed the database tables with the necessary app configs.
- Run `docker-compose up --build` to build the docker image and spin up all the necessary services
- As an alternative to running the app on docker, you can run `npm run start:dev`

###  PATCH
-`http://localhost:3000/v1.0/api/ingredient/{id}` 
- make sure an admin is already logged in with his/her token in the cookies.
- object
```
{
 "name": "tomato",
 "unit": "6 grams"
{
```
*only works if `tomato` exists in database. 
### DELETE 
-`http://localhost:3000/v1.0/api/ingredient/{id}` 
- make sure an admin is already logged in with his/her token in the cookies.

*only works if `tomato` exists in database. 
### GET
-`http://localhost:3000/v1.0/api/ingredient/all` 
- make sure an admin is already logged in with his/her token in the cookies.

### CODE REVIEW

**Comment by dchima** [comment]
[Admin can get/update/delete ingredients](https://trello.com/c/oxCq0OpA/25-admin-can-get-update-delete-ingredients)

File: src/routes/ingredient.js [hunk 1]
```diff
@@ -7,10 +7,13 @@ const router = Router();
...
 const { authenticate, isVerified } = AuthMiddleware;
 const { verifyRoles } = UserMiddleware;
-const { onAddIngredient } = IngredientMiddleware;
-const { addIngredient } = IngredientController;
+const { onAddIngredient, ingredientCheck } = IngredientMiddleware;
+const { addIngredient, updateIngredient, deleteIngredient, getAllIngredients } = IngredientController;
 const { admin } = PermissionsData;
 
 router.post('/', authenticate, verifyRoles(admin), isVerified, onAddIngredient, addIngredient);
+router.patch('/edit', authenticate, verifyRoles(admin), isVerified, ingredientCheck, updateIngredient);
```
[the remark this answers is not in the document]
Comment (EziamakaNV) [reply src/routes/ingredient.js]: The same for the get route as well. A request to ...ingredient/ should get everything, if a particular ingredient is needed then we can have ...ingredient/:Id or ingredient/:name

File: src/validations/ingredientValidation.js [hunk 2]
```diff
@@ -25,4 +25,21 @@ export default class IngredientValidations {
...
   }
+
+  /**
+   * object for updating ingredients
+   * @param {object} payload
+   * @returns {object | boolean} - returns error object or validation boolean
+   */
+  static validateIngredient(payload) {
+    const schema = {
+      name: joi.string().min(3).max(30)
```
[the remark this answers is not in the document]
Comment (dchima) [reply src/validations/ingredientValidation.js]: The schema you wrote are required. checks for edid routes should not be required, or else you'd have to send all parameters to the database for every update.

File: src/routes/ingredient.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dchima) [reply src/routes/ingredient.js]: this was the default. the issue was that on swagger docs it didn't recognise duplicate routes. So I decided to make them unique by adding explicit parameters.

**Comment by dchima** [comment]
@EziamakaNV I realize now that we did not follow the database setup we agreed on in the beginning. The `unit` column should simply be a string of the ingredient units, eg `grams`, `kilograms`, `litres`... *without the number.*
This is because the mealsIngredient table will already have a `quantity` attribute that references the units.

**Comment by dchima** [comment]
https://trello.com/c/KubHgWb0

**Pull request merged**